<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="taxize">
<title>Introduction to taxize • taxize</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to taxize">
<meta property="og:description" content="taxize">
<meta property="og:image" content="https://docs.ropensci.org/taxize/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">taxize</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.9.100</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="active nav-item">
  <a class="nav-link" href="../articles/taxize.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/case_study.html">Case study - resolving Species Plantarum names</a>
    <a class="dropdown-item" href="../articles/datasources.html">taxize data sources</a>
    <a class="dropdown-item" href="../articles/infotable.html">Key functions in taxize</a>
    <a class="dropdown-item" href="../articles/name_cleaning.html">Strategies for programmatic name cleaning</a>
    <a class="dropdown-item" href="../articles/taxonomic_ranks.html">taxonomic ranks</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/taxize/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to taxize</h1>
                        <h4 data-toc-skip class="author">Scott Chamberlain</h4>
            
            <h4 data-toc-skip class="date">2020-09-17</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/taxize/blob/HEAD/vignettes/taxize.Rmd" class="external-link"><code>vignettes/taxize.Rmd</code></a></small>
      <div class="d-none name"><code>taxize.Rmd</code></div>
    </div>

    
    
<p><code>taxize</code> is a taxonomic toolbelt for R. <code>taxize</code> wraps APIs for a large suite of taxonomic databases availab on the web.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>First, install and load <code>taxize</code> into the R session.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"taxize"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://docs.ropensci.org/taxize/">"taxize"</a></span><span class="op">)</span></span></code></pre></div>
<p>Advanced users can also download and install the latest development copy from GitHub (<a href="https://github.com/ropensci/taxize" class="external-link uri">https://github.com/ropensci/taxize</a>)</p>
</div>
<div class="section level2">
<h2 id="resolve-taxonomic-name">Resolve taxonomic name<a class="anchor" aria-label="anchor" href="#resolve-taxonomic-name"></a>
</h2>
<p>This is a common task in biology. We often have a list of species names and we want to know a) if we have the most up to date names, b) if our names are spelled correctly, and c) the scientific name for a common name. One way to resolve names is via the Global Names Resolver (GNR) service provided by the Encyclopedia of Life. Here, we are searching for two misspelled names:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gnr_resolve.html">gnr_resolve</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Helianthos annus"</span>, <span class="st">"Homo saapiens"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 x 5</span></span>
<span><span class="co">#&gt;   user_supplied_name submitted_name  matched_name     data_source_title    score</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;              &lt;chr&gt;           &lt;chr&gt;            &lt;chr&gt;                &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1 Helianthos annus   Helianthos ann… Helianthus annus uBio NameBank         0.75</span></span>
<span><span class="co">#&gt; 2 Helianthos annus   Helianthos ann… Helianthus annu… Catalogue of Life     0.75</span></span>
<span><span class="co">#&gt; 3 Helianthos annus   Helianthos ann… Helianthus annu… ITIS                  0.75</span></span>
<span><span class="co">#&gt; 4 Helianthos annus   Helianthos ann… Helianthus annu… NCBI                  0.75</span></span>
<span><span class="co">#&gt; 5 Helianthos annus   Helianthos ann… Helianthus annu… GRIN Taxonomy for P…  0.75</span></span>
<span><span class="co">#&gt; 6 Helianthos annus   Helianthos ann… Helianthus annu… Union 4               0.75</span></span></code></pre></div>
<p>The correct spellings are <em>Helianthus annuus</em> and <em>Homo sapiens</em>.</p>
<p>taxize takes the approach that the user should be able to make decisions about what resource to trust, rather than making the decision. The GNR service provides data from a variety of data sources. The user may trust a specific data source, thus may want to use the names from that data source. In the future, we may provide the ability for taxize to suggest the best match from a variety of sources.</p>
<p>Another common use case is when there are many synonyms for a species. In this example, we have three synonyms of the currently accepted name for a species.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>mynames <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Helianthus annuus ssp. jaegeri"</span>, <span class="st">"Helianthus annuus ssp. lenticularis"</span>, <span class="st">"Helianthus annuus ssp. texanus"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>(tsn <span class="ot">&lt;-</span> <span class="fu">get_tsn</span>(mynames, <span class="at">accepted =</span> <span class="cn">FALSE</span>))</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>══  <span class="dv">3</span> queries  ═══════════════</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>✔  Found<span class="sc">:</span>  Helianthus annuus ssp. jaegeri</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>✔  Found<span class="sc">:</span>  Helianthus annuus ssp. lenticularis</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>✔  Found<span class="sc">:</span>  Helianthus annuus ssp. texanus</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>══  Results  ═════════════════</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>● Total<span class="sc">:</span> <span class="dv">3</span> </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>● Found<span class="sc">:</span> <span class="dv">3</span> </span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>● Not Found<span class="sc">:</span> <span class="dv">0</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"525928"</span> <span class="st">"525929"</span> <span class="st">"525930"</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(,<span class="st">"class"</span>)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"tsn"</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(,<span class="st">"match"</span>)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"found"</span> <span class="st">"found"</span> <span class="st">"found"</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(,<span class="st">"multiple_matches"</span>)</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="cn">FALSE</span> <span class="cn">FALSE</span> <span class="cn">FALSE</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(,<span class="st">"pattern_match"</span>)</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="cn">FALSE</span> <span class="cn">FALSE</span> <span class="cn">FALSE</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(,<span class="st">"uri"</span>)</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"https://www.itis.gov/servlet/SingleRpt/SingleRpt?search_topic=TSN&amp;search_value=525928"</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>[<span class="dv">2</span>] <span class="st">"https://www.itis.gov/servlet/SingleRpt/SingleRpt?search_topic=TSN&amp;search_value=525929"</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>[<span class="dv">3</span>] <span class="st">"https://www.itis.gov/servlet/SingleRpt/SingleRpt?search_topic=TSN&amp;search_value=525930"</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(tsn, itis_acceptname)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>[[<span class="dv">1</span>]]</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  submittedtsn      acceptedname acceptedtsn author</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span>       <span class="dv">525928</span> Helianthus annuus       <span class="dv">36616</span>     L.</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>[[<span class="dv">2</span>]]</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>  submittedtsn      acceptedname acceptedtsn author</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span>       <span class="dv">525929</span> Helianthus annuus       <span class="dv">36616</span>     L.</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>[[<span class="dv">3</span>]]</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  submittedtsn      acceptedname acceptedtsn author</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span>       <span class="dv">525930</span> Helianthus annuus       <span class="dv">36616</span>     L.</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="retrieve-higher-taxonomic-names">Retrieve higher taxonomic names<a class="anchor" aria-label="anchor" href="#retrieve-higher-taxonomic-names"></a>
</h2>
<p>Another task biologists often face is getting higher taxonomic names for a taxa list. Having the higher taxonomy allows you to put into context the relationships of your species list. For example, you may find out that species A and species B are in Family C, which may lead to some interesting insight, as opposed to not knowing that Species A and B are closely related. This also makes it easy to aggregate/standardize data to a specific taxonomic level (e.g., family level) or to match data to other databases with different taxonomic resolution (e.g., trait databases).</p>
<p>A number of data sources in taxize provide the capability to retrieve higher taxonomic names, but we will highlight two of the more useful ones: Integrated Taxonomic Information System (ITIS) and National Center for Biotechnology Information (NCBI). First, we’ll search for two species, <em>Abies procera} and </em>Pinus contorta* within ITIS.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">specieslist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Abies procera"</span>,<span class="st">"Pinus contorta"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/classification.html">classification</a></span><span class="op">(</span><span class="va">specieslist</span>, db <span class="op">=</span> <span class="st">'itis'</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  2 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  Abies procera</span></span>
<span><span class="co">#&gt; ✔  Found:  Pinus contorta</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 2 </span></span>
<span><span class="co">#&gt; ● Found: 2 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="co">#&gt; $`Abies procera`</span></span>
<span><span class="co">#&gt;               name          rank     id</span></span>
<span><span class="co">#&gt; 1          Plantae       kingdom 202422</span></span>
<span><span class="co">#&gt; 2    Viridiplantae    subkingdom 954898</span></span>
<span><span class="co">#&gt; 3     Streptophyta  infrakingdom 846494</span></span>
<span><span class="co">#&gt; 4      Embryophyta superdivision 954900</span></span>
<span><span class="co">#&gt; 5     Tracheophyta      division 846496</span></span>
<span><span class="co">#&gt; 6  Spermatophytina   subdivision 846504</span></span>
<span><span class="co">#&gt; 7        Pinopsida         class 500009</span></span>
<span><span class="co">#&gt; 8          Pinidae      subclass 954916</span></span>
<span><span class="co">#&gt; 9          Pinales         order 500028</span></span>
<span><span class="co">#&gt; 10        Pinaceae        family  18030</span></span>
<span><span class="co">#&gt; 11           Abies         genus  18031</span></span>
<span><span class="co">#&gt; 12   Abies procera       species 181835</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`Pinus contorta`</span></span>
<span><span class="co">#&gt;               name          rank     id</span></span>
<span><span class="co">#&gt; 1          Plantae       kingdom 202422</span></span>
<span><span class="co">#&gt; 2    Viridiplantae    subkingdom 954898</span></span>
<span><span class="co">#&gt; 3     Streptophyta  infrakingdom 846494</span></span>
<span><span class="co">#&gt; 4      Embryophyta superdivision 954900</span></span>
<span><span class="co">#&gt; 5     Tracheophyta      division 846496</span></span>
<span><span class="co">#&gt; 6  Spermatophytina   subdivision 846504</span></span>
<span><span class="co">#&gt; 7        Pinopsida         class 500009</span></span>
<span><span class="co">#&gt; 8          Pinidae      subclass 954916</span></span>
<span><span class="co">#&gt; 9          Pinales         order 500028</span></span>
<span><span class="co">#&gt; 10        Pinaceae        family  18030</span></span>
<span><span class="co">#&gt; 11           Pinus         genus  18035</span></span>
<span><span class="co">#&gt; 12  Pinus contorta       species 183327</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "classification"</span></span>
<span><span class="co">#&gt; attr(,"db")</span></span>
<span><span class="co">#&gt; [1] "itis"</span></span></code></pre></div>
<p>It turns out both species are in the family Pinaceae. You can also get this type of information from the NCBI by doing <code>classification(specieslist, db = 'ncbi')</code>.</p>
<p>Instead of a full classification, you may only want a single name, say a family name for your species of interest. The function <code>tax_name</code> is built just for this purpose. As with the <code>classification</code> function you can specify the data source with the <code>db</code> argument, either ITIS or NCBI.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tax_name.html">tax_name</a></span><span class="op">(</span><span class="st">"Helianthus annuus"</span>, get <span class="op">=</span> <span class="st">"family"</span>, db <span class="op">=</span> <span class="st">"ncbi"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  1 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  Helianthus+annuus</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 1 </span></span>
<span><span class="co">#&gt; ● Found: 1 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="co">#&gt;     db             query     family</span></span>
<span><span class="co">#&gt; 1 ncbi Helianthus annuus Asteraceae</span></span></code></pre></div>
<p>It may happen that a data source does not provide information on the queried species, than one could take the result from another source and union the results from the different sources.</p>
</div>
<div class="section level2">
<h2 id="interactive-name-selection">Interactive name selection<a class="anchor" aria-label="anchor" href="#interactive-name-selection"></a>
</h2>
<p>As mentioned most databases use a numeric code to reference a species. A general workflow in taxize is: Retrieve Code for the queried species and then use this code to query more data/information.</p>
<p>Below are a few examples. When you run these examples in R, you are presented with a command prompt asking for the row that contains the name you would like back; that output is not printed below for brevity. In this example, the search term has many matches. The function returns a data frame of the matches, and asks for the user to input what row number to accept.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_uid.html">get_uid</a></span><span class="op">(</span><span class="st">"Pinus"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  1 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  Pinus</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 1 </span></span>
<span><span class="co">#&gt; ● Found: 1 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="co">#&gt; [1] "3337"</span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "uid"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"uri")</span></span>
<span><span class="co">#&gt; [1] "https://www.ncbi.nlm.nih.gov/taxonomy/3337"</span></span></code></pre></div>
<p>In another example, you can pass in a long character vector of taxonomic names (although this one is rather short for demo purposes):</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">splist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"annona cherimola"</span>, <span class="st">'annona muricata'</span>, <span class="st">"quercus robur"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/get_tsn.html">get_tsn</a></span><span class="op">(</span><span class="va">splist</span>, searchtype <span class="op">=</span> <span class="st">"scientific"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  3 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  annona cherimola</span></span>
<span><span class="co">#&gt; ✔  Found:  annona muricata</span></span>
<span><span class="co">#&gt; ✔  Found:  quercus robur</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 3 </span></span>
<span><span class="co">#&gt; ● Found: 3 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="co">#&gt; [1] "506198" "18098"  "19405" </span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "tsn"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "found" "found" "found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] FALSE FALSE  TRUE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] FALSE FALSE  TRUE</span></span>
<span><span class="co">#&gt; attr(,"uri")</span></span>
<span><span class="co">#&gt; [1] "https://www.itis.gov/servlet/SingleRpt/SingleRpt?search_topic=TSN&amp;search_value=506198"</span></span>
<span><span class="co">#&gt; [2] "https://www.itis.gov/servlet/SingleRpt/SingleRpt?search_topic=TSN&amp;search_value=18098" </span></span>
<span><span class="co">#&gt; [3] "https://www.itis.gov/servlet/SingleRpt/SingleRpt?search_topic=TSN&amp;search_value=19405"</span></span></code></pre></div>
<p>There are functions for many other sources</p>
<ul>
<li><code><a href="../reference/get_boldid.html">get_boldid()</a></code></li>
<li><code><a href="../reference/get_eolid.html">get_eolid()</a></code></li>
<li><code><a href="../reference/get_gbifid.html">get_gbifid()</a></code></li>
<li><code><a href="../reference/get_nbnid.html">get_nbnid()</a></code></li>
<li><code><a href="../reference/get_tpsid.html">get_tpsid()</a></code></li>
</ul>
<p>Sometimes with these functions you get a lot of data back. In these cases you may want to limit your choices. Soon we will incorporate the ability to filter using <code>regex</code> to limit matches, but for now, we have a new parameter, <code>rows</code>, which lets you select certain rows. For example, you can select the first row of each given name, which means there is no interactive component:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_nbnid.html">get_nbnid</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Zootoca vivipara"</span>,<span class="st">"Pinus contorta"</span><span class="op">)</span>, rows <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  2 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  Zootoca vivipara</span></span>
<span><span class="co">#&gt; ✔  Found:  Pinus contorta</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 2 </span></span>
<span><span class="co">#&gt; ● Found: 2 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="co">#&gt; [1] "NHMSYS0001706186" "NBNSYS0000004786"</span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "nbnid"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "found" "found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] TRUE TRUE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] FALSE FALSE</span></span>
<span><span class="co">#&gt; attr(,"uri")</span></span>
<span><span class="co">#&gt; [1] "https://species.nbnatlas.org/species/NHMSYS0001706186"</span></span>
<span><span class="co">#&gt; [2] "https://species.nbnatlas.org/species/NBNSYS0000004786"</span></span></code></pre></div>
<p>Or you can select a range of rows</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_nbnid.html">get_nbnid</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Zootoca vivipara"</span>,<span class="st">"Pinus contorta"</span><span class="op">)</span>, rows <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  2 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  Zootoca vivipara</span></span>
<span><span class="co">#&gt; ✔  Found:  Pinus contorta</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 2 </span></span>
<span><span class="co">#&gt; ● Found: 2 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="co">#&gt; [1] "NHMSYS0001706186" "NBNSYS0000004786"</span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "nbnid"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "found" "found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] TRUE TRUE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] TRUE TRUE</span></span>
<span><span class="co">#&gt; attr(,"uri")</span></span>
<span><span class="co">#&gt; [1] "https://species.nbnatlas.org/species/NHMSYS0001706186"</span></span>
<span><span class="co">#&gt; [2] "https://species.nbnatlas.org/species/NBNSYS0000004786"</span></span></code></pre></div>
<p>In addition, in case you don’t want to do interactive name selection in the case where there are a lot of names, you can get all data back with functions of the form, e.g., <code><a href="../reference/get_tsn.html">get_tsn_()</a></code>, and likewise for other data sources. For example:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_nbnid.html">get_nbnid_</a></span><span class="op">(</span><span class="st">"Poa annua"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">NROW</a></span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">`Poa annua`</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 25</span></span></code></pre></div>
<p>That’s a lot of data, so we can get only certain rows back</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_nbnid.html">get_nbnid_</a></span><span class="op">(</span><span class="st">"Poa annua"</span>, rows <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`Poa annua`</span></span>
<span><span class="co">#&gt;                guid     scientificName    rank taxonomicStatus</span></span>
<span><span class="co">#&gt; 1  NBNSYS0000002544          Poa annua species        accepted</span></span>
<span><span class="co">#&gt; 2  NBNSYS0200001901       Bellis annua species        accepted</span></span>
<span><span class="co">#&gt; 3  NBNSYS0200003392   Triumfetta annua species        accepted</span></span>
<span><span class="co">#&gt; 4  NBNSYS0200002555        Lonas annua species        accepted</span></span>
<span><span class="co">#&gt; 5  NHMSYS0000456951  Carrichtera annua species        accepted</span></span>
<span><span class="co">#&gt; 6  NHMSYS0000461807 Poa labillardierei species        accepted</span></span>
<span><span class="co">#&gt; 7  NHMSYS0000461808      Poa ligularis species        accepted</span></span>
<span><span class="co">#&gt; 8  NHMSYS0000461817     Poa sieberiana species        accepted</span></span>
<span><span class="co">#&gt; 9  NHMSYS0000461805         Poa gunnii species        accepted</span></span>
<span><span class="co">#&gt; 10 NHMSYS0000461801     Poa costiniana species        accepted</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="coerce-numericsalphanumerics-to-taxon-ids">Coerce numerics/alphanumerics to taxon IDs<a class="anchor" aria-label="anchor" href="#coerce-numericsalphanumerics-to-taxon-ids"></a>
</h2>
<p>We’ve also introduced in <code>v0.5</code> the ability to coerce numerics and alphanumerics to taxonomic ID classes that are usually only retrieved via <code>get_*()</code> functions.</p>
<p>For example, adfafd</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_gbifid.html">as.gbifid</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_gbifid.html">get_gbifid</a></span><span class="op">(</span><span class="st">"Poa annua"</span><span class="op">)</span><span class="op">)</span> <span class="co"># already a uid, returns the same</span></span>
<span><span class="co">#&gt; ══  1 queries  ═══════════════</span></span>
<span><span class="co">#&gt;    gbifid             scientificname    rank   status matchtype</span></span>
<span><span class="co">#&gt; 1 2704179               Poa annua L. species ACCEPTED     EXACT</span></span>
<span><span class="co">#&gt; 2 8422205 Poa annua Cham. &amp; Schltdl. species  SYNONYM     EXACT</span></span>
<span><span class="co">#&gt; 3 7730008           Poa annua Steud. species DOUBTFUL     EXACT</span></span>
<span><span class="co">#&gt; ✖  Not Found:  Poa annua</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 1 </span></span>
<span><span class="co">#&gt; ● Found: 0 </span></span>
<span><span class="co">#&gt; ● Not Found: 1</span></span>
<span><span class="co">#&gt; [1] NA</span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "gbifid"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "not found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="fu"><a href="../reference/get_gbifid.html">as.gbifid</a></span><span class="op">(</span><span class="fl">2704179</span><span class="op">)</span> <span class="co"># numeric</span></span>
<span><span class="co">#&gt; [1] "2704179"</span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "gbifid"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"uri")</span></span>
<span><span class="co">#&gt; [1] "https://www.gbif.org/species/2704179"</span></span>
<span><span class="fu"><a href="../reference/get_gbifid.html">as.gbifid</a></span><span class="op">(</span><span class="st">"2704179"</span><span class="op">)</span> <span class="co"># character</span></span>
<span><span class="co">#&gt; [1] "2704179"</span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "gbifid"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"uri")</span></span>
<span><span class="co">#&gt; [1] "https://www.gbif.org/species/2704179"</span></span>
<span><span class="fu"><a href="../reference/get_gbifid.html">as.gbifid</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"2704179"</span>,<span class="st">"2435099"</span>,<span class="st">"3171445"</span><span class="op">)</span><span class="op">)</span> <span class="co"># list, either numeric or character</span></span>
<span><span class="co">#&gt; [1] "2704179" "2435099" "3171445"</span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "gbifid"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "found" "found" "found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt; attr(,"uri")</span></span>
<span><span class="co">#&gt; [1] "https://www.gbif.org/species/2704179"</span></span>
<span><span class="co">#&gt; [2] "https://www.gbif.org/species/2435099"</span></span>
<span><span class="co">#&gt; [3] "https://www.gbif.org/species/3171445"</span></span></code></pre></div>
<p>These <code>as.*()</code> functions do a quick check of the web resource to make sure it’s a real ID. However, you can turn this check off, making this coercion much faster:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">replicate</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fu"><a href="../reference/get_gbifid.html">as.gbifid</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"2704179"</span>,<span class="st">"2435099"</span>,<span class="st">"3171445"</span><span class="op">)</span>, check<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="co">#&gt;    user  system elapsed </span></span>
<span><span class="co">#&gt;   0.092   0.003   4.850</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">replicate</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fu"><a href="../reference/get_gbifid.html">as.gbifid</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"2704179"</span>,<span class="st">"2435099"</span>,<span class="st">"3171445"</span><span class="op">)</span>, check<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="co">#&gt;    user  system elapsed </span></span>
<span><span class="co">#&gt;   0.002   0.000   0.002</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="what-taxa-are-downstream-of-my-taxon-of-interest">What taxa are downstream of my taxon of interest?<a class="anchor" aria-label="anchor" href="#what-taxa-are-downstream-of-my-taxon-of-interest"></a>
</h2>
<p>If someone is not a taxonomic specialist on a particular taxon he likely does not know what children taxa are within a family, or within a genus. This task becomes especially unwieldy when there are a large number of taxa downstream. You can of course go to a website like Wikispecies or Encyclopedia of Life to get downstream names. However, taxize provides an easy way to programatically search for downstream taxa for the Integrated Taxonomic Information System.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">apis_itis_id</span> <span class="op">&lt;-</span> <span class="fl">154395</span> <span class="co"># id for Apis, fetched beforehand to save time here</span></span>
<span><span class="fu"><a href="../reference/downstream.html">downstream</a></span><span class="op">(</span><span class="va">apis_itis_id</span>, downto <span class="op">=</span> <span class="st">"species"</span>, db <span class="op">=</span> <span class="st">"itis"</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`154395`</span></span>
<span><span class="co">#&gt;       tsn parentname parenttsn rankname          taxonname rankid</span></span>
<span><span class="co">#&gt; 1 1128092       Apis    154395  species     Apis laboriosa    220</span></span>
<span><span class="co">#&gt; 2  154396       Apis    154395  species     Apis mellifera    220</span></span>
<span><span class="co">#&gt; 3  763550       Apis    154395  species Apis andreniformis    220</span></span>
<span><span class="co">#&gt; 4  763551       Apis    154395  species        Apis cerana    220</span></span>
<span><span class="co">#&gt; 5  763552       Apis    154395  species       Apis dorsata    220</span></span>
<span><span class="co">#&gt; 6  763553       Apis    154395  species        Apis florea    220</span></span>
<span><span class="co">#&gt; 7  763554       Apis    154395  species Apis koschevnikovi    220</span></span>
<span><span class="co">#&gt; 8  763555       Apis    154395  species   Apis nigrocincta    220</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "downstream"</span></span>
<span><span class="co">#&gt; attr(,"db")</span></span>
<span><span class="co">#&gt; [1] "itis"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="direct-children">Direct children<a class="anchor" aria-label="anchor" href="#direct-children"></a>
</h2>
<p>You may sometimes only want the direct children. We got you covered on that front, with methods for ITIS and NCBI.</p>
<p>The direct children (genera in this case) of <em>Pinaceae</em> using NCBI data:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/children.html">children</a></span><span class="op">(</span><span class="st">"Pinaceae"</span>, db <span class="op">=</span> <span class="st">"ncbi"</span><span class="op">)</span></span>
<span><span class="co">#&gt; $Pinaceae</span></span>
<span><span class="co">#&gt;    childtaxa_id childtaxa_name childtaxa_rank</span></span>
<span><span class="co">#&gt; 1        123600     Nothotsuga          genus</span></span>
<span><span class="co">#&gt; 2         64685        Cathaya          genus</span></span>
<span><span class="co">#&gt; 3          3358          Tsuga          genus</span></span>
<span><span class="co">#&gt; 4          3356    Pseudotsuga          genus</span></span>
<span><span class="co">#&gt; 5          3354    Pseudolarix          genus</span></span>
<span><span class="co">#&gt; 6          3337          Pinus          genus</span></span>
<span><span class="co">#&gt; 7          3328          Picea          genus</span></span>
<span><span class="co">#&gt; 8          3325          Larix          genus</span></span>
<span><span class="co">#&gt; 9          3323     Keteleeria          genus</span></span>
<span><span class="co">#&gt; 10         3321         Cedrus          genus</span></span>
<span><span class="co">#&gt; 11         3319          Abies          genus</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "children"</span></span>
<span><span class="co">#&gt; attr(,"db")</span></span>
<span><span class="co">#&gt; [1] "ncbi"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get-ncbi-id-from-genbank-ids">Get NCBI ID from GenBank Ids<a class="anchor" aria-label="anchor" href="#get-ncbi-id-from-genbank-ids"></a>
</h2>
<p>With accession numbers</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/genbank2uid.html">genbank2uid</a></span><span class="op">(</span>id <span class="op">=</span> <span class="st">'AJ748748'</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; [1] "282199"</span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "uid"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"uri")</span></span>
<span><span class="co">#&gt; [1] "https://www.ncbi.nlm.nih.gov/taxonomy/282199"</span></span>
<span><span class="co">#&gt; attr(,"name")</span></span>
<span><span class="co">#&gt; [1] "Nereida ignava 16S rRNA gene, type strain 2SM4T"</span></span></code></pre></div>
<p>With gi numbers</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/genbank2uid.html">genbank2uid</a></span><span class="op">(</span>id <span class="op">=</span> <span class="fl">62689767</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; [1] "282199"</span></span>
<span><span class="co">#&gt; attr(,"class")</span></span>
<span><span class="co">#&gt; [1] "uid"</span></span>
<span><span class="co">#&gt; attr(,"match")</span></span>
<span><span class="co">#&gt; [1] "found"</span></span>
<span><span class="co">#&gt; attr(,"multiple_matches")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"pattern_match")</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; attr(,"uri")</span></span>
<span><span class="co">#&gt; [1] "https://www.ncbi.nlm.nih.gov/taxonomy/282199"</span></span>
<span><span class="co">#&gt; attr(,"name")</span></span>
<span><span class="co">#&gt; [1] "Nereida ignava 16S rRNA gene, type strain 2SM4T"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="matching-species-tables-with-different-taxonomic-resolution">Matching species tables with different taxonomic resolution<a class="anchor" aria-label="anchor" href="#matching-species-tables-with-different-taxonomic-resolution"></a>
</h2>
<p>Biologist often need to match different sets of data tied to species. For example, trait-based approaches are a promising tool in ecology. One problem is that abundance data must be matched with trait databases. These two data tables may contain species information on different taxonomic levels and possibly data must be aggregated to a joint taxonomic level, so that the data can be merged. taxize can help in this data-cleaning step, providing a reproducible workflow:</p>
<p>We can use the mentioned <code>classification</code>-function to retrieve the taxonomic hierarchy and then search the hierarchies up- and downwards for matches. Here is an example to match a species with names on three different taxonomic levels.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">A</span> <span class="op">&lt;-</span> <span class="st">"gammarus roeseli"</span></span>
<span></span>
<span><span class="va">B1</span> <span class="op">&lt;-</span> <span class="st">"gammarus roeseli"</span></span>
<span><span class="va">B2</span> <span class="op">&lt;-</span> <span class="st">"gammarus"</span></span>
<span><span class="va">B3</span> <span class="op">&lt;-</span> <span class="st">"gammaridae"</span></span>
<span></span>
<span><span class="va">A_clas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/classification.html">classification</a></span><span class="op">(</span><span class="va">A</span>, db <span class="op">=</span> <span class="st">'ncbi'</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  1 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  gammarus+roeseli</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 1 </span></span>
<span><span class="co">#&gt; ● Found: 1 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="va">B1_clas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/classification.html">classification</a></span><span class="op">(</span><span class="va">B1</span>, db <span class="op">=</span> <span class="st">'ncbi'</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  1 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  gammarus+roeseli</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 1 </span></span>
<span><span class="co">#&gt; ● Found: 1 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="va">B2_clas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/classification.html">classification</a></span><span class="op">(</span><span class="va">B2</span>, db <span class="op">=</span> <span class="st">'ncbi'</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  1 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  gammarus</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 1 </span></span>
<span><span class="co">#&gt; ● Found: 1 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span><span class="va">B3_clas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/classification.html">classification</a></span><span class="op">(</span><span class="va">B3</span>, db <span class="op">=</span> <span class="st">'ncbi'</span><span class="op">)</span></span>
<span><span class="co">#&gt; ══  1 queries  ═══════════════</span></span>
<span><span class="co">#&gt; ✔  Found:  gammaridae</span></span>
<span><span class="co">#&gt; ══  Results  ═════════════════</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ● Total: 1 </span></span>
<span><span class="co">#&gt; ● Found: 1 </span></span>
<span><span class="co">#&gt; ● Not Found: 0</span></span>
<span></span>
<span><span class="va">B1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">A</span>, <span class="va">B1</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; [1] "gammarus roeseli"</span></span>
<span><span class="va">A_clas</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">rank</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="va">A_clas</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">name</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">B2</span><span class="op">]</span></span>
<span><span class="co">#&gt; [1] "genus"</span></span>
<span><span class="va">A_clas</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">rank</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="va">A_clas</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">name</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">B3</span><span class="op">]</span></span>
<span><span class="co">#&gt; [1] "family"</span></span></code></pre></div>
<p>If we find a direct match (here <em>Gammarus roeseli</em>), we are lucky. But we can also match Gammaridae with <em>Gammarus roeseli</em>, but on a lower taxonomic level. A more comprehensive and realistic example (matching a trait table with an abundance table) is given in the vignette on matching.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
